google-cloud-clients/google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/PublisherImplTest.java

```diff
@@ -241,6 +243,177 @@ public void testPublishMixedSizeAndDuration() throws Exception {
         PubsubMessage.newBuilder().setData(ByteString.copyFromUtf8(data)).build());
   }
 
+  @Test
+  public void testBatchedMessagesWithOrderingKeyByNum() throws Exception {
+    // Limit the number of maximum elements in a single batch to 3.
+    Publisher publisher =
+        getTestPublisherBuilder()
+            .setBatchingSettings(
+                Publisher.Builder.DEFAULT_BATCHING_SETTINGS
+                    .toBuilder()
+                    .setElementCountThreshold(3L)
+                    .setDelayThreshold(Duration.ofSeconds(100))
+                    .build())
+            .setEnableMessageOrdering(true)
+            .build();
+    testPublisherServiceImpl.setAutoPublishResponse(true);
+
+    // Publish two messages with ordering key, "OrderA", and other two messages with "OrderB".
+    ApiFuture<String> publishFuture1 = sendTestMessageWithOrderingKey(publisher, "m1", "OrderA");
+    ApiFuture<String> publishFuture2 = sendTestMessageWithOrderingKey(publisher, "m2", "OrderB");
+    ApiFuture<String> publishFuture3 = sendTestMessageWithOrderingKey(publisher, "m3", "OrderA");
+    ApiFuture<String> publishFuture4 = sendTestMessageWithOrderingKey(publisher, "m4", "OrderB");
+
+    // Verify that none of them were published since the batching size is 3.
+    assertFalse(publishFuture1.isDone());
+    assertFalse(publishFuture2.isDone());
+    assertFalse(publishFuture3.isDone());
+    assertFalse(publishFuture4.isDone());
+
+    // One of the batches reaches the limit.
+    ApiFuture<String> publishFuture5 = sendTestMessageWithOrderingKey(publisher, "m5", "OrderA");
+    // Verify that they were delivered in order per ordering key.
+    assertTrue(Integer.parseInt(publishFuture1.get()) < Integer.parseInt(publishFuture3.get()));
+    assertTrue(Integer.parseInt(publishFuture2.get()) < Integer.parseInt(publishFuture4.get()));
+    assertTrue(Integer.parseInt(publishFuture3.get()) < Integer.parseInt(publishFuture5.get()));
+
```

Comment:
I think correct check should be Integer.parseInt(publishFuture5.get()) < Integer.parseInt(publishFuture2.get()) if batch with ordering key "OrderA" is sent first or Integer.parseInt(publishFuture2.get()) < Integer.parseInt(publishFuture1.get()) if batch with orderking key "OrderB" is sent first?